packages/jest-mock/src/index.ts

```diff
@@ -1408,14 +1442,15 @@ export class ModuleMocker {
   }
 
   resetAllMocks(): void {
-    this._spyState.forEach(reset => reset());
+    this.clearAllMocks();
     this._mockConfigRegistry = new WeakMap();
-    this._mockState = new WeakMap();
+    this._spyState.forEach(spyState => spyState.reset?.());
   }
 
   restoreAllMocks(): void {
-    this._mockState = new WeakMap();
-    this._spyState.forEach(restore => restore());
+    this.clearAllMocks();
```

Comment:
Other idea was to call `this.resetAllMocks()` here instead of `this.clearAllMocks()`. Looks like resetting implementation in this case is just waste of time, because that implementation will be discarded I next lines.